### PR TITLE
refactor(ES6): upgrade AMDDefineDependencyParserPlugin to ES6

### PR DESCRIPTION
**Pull request by shubheksha**
**What kind of change does this PR introduce?**
refactor, ES5 => ES6

**Did you add tests for your changes?**
All existing tests pass

**If relevant, link to documentation update:**
N/A

**Summary**
upgrade AMDDefineDependencyParserPlugin to ES6

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: lib/dependencies/AMDDefineDependencyParserPlugin.js [hunk 1]
```diff
@@ -21,220 +23,221 @@ function isBoundFunctionExpression(expr) {
...
-					namedModule = expr.arguments[0].value;
-					// define("...", ...)
-					if(expr.arguments[1].type === "FunctionExpression" || isBoundFunctionExpression(expr.arguments[1])) {
-						// define("...", f() {...})
-						fn = expr.arguments[1];
-					} else if(expr.arguments[1].type === "ObjectExpression") {
-						// define("...", {...})
-						obj = expr.arguments[1];
+	apply(parser) {
+		const options = parser.options;
```
Comment (sokra) [inline lib/dependencies/AMDDefineDependencyParserPlugin.js]: `this.options`

File: lib/dependencies/AMDDefineDependencyParserPlugin.js [hunk 2]
```diff
@@ -21,220 +23,221 @@ function isBoundFunctionExpression(expr) {
...
+		parser.plugin("call define:amd:array", (expr, param, identifiers, namedModule) => {
+			if(param.isArray()) {
+				param.items.forEach((param, idx) => {
+					if(param.isString() && ["require", "module", "exports"].indexOf(param.string) >= 0)
+						identifiers[idx] = param.string;
+					const result = parser.applyPluginsBailResult("call define:amd:item", expr, param, namedModule);
+					if(result === undefined) {
+						parser.applyPluginsBailResult("call define:amd:context", expr, param);
+					}
+				}, parser);
```
Comment (sokra) [inline lib/dependencies/AMDDefineDependencyParserPlugin.js]: context argument not needed

File: lib/dependencies/AMDDefineDependencyParserPlugin.js [hunk 3]
```diff
@@ -21,220 +23,221 @@ function isBoundFunctionExpression(expr) {
...
+						dep.loc = expr.loc;
+						parser.state.current.addDependency(dep);
+					} else {
+						dep = new AMDRequireItemDependency(request);
+						dep.loc = expr.loc;
+						dep.optional = !!parser.scope.inTry;
+						parser.state.current.addDependency(dep);
+					}
+					deps.push(dep);
+				}, parser);
```
Comment (sokra) [inline lib/dependencies/AMDDefineDependencyParserPlugin.js]: context arg not needed

File: lib/dependencies/AMDDefineDependencyParserPlugin.js [hunk 4]
```diff
@@ -21,220 +23,221 @@ function isBoundFunctionExpression(expr) {
...
+			}
+		});
+		parser.plugin("call define:amd:item", (expr, param, namedModule) => {
+			if(param.isConditional()) {
+				param.options.forEach((param) => {
+					const result = parser.applyPluginsBailResult("call define:amd:item", expr, param);
+					if(result === undefined) {
+						parser.applyPluginsBailResult("call define:amd:context", expr, param);
+					}
+				}, parser);
```
Comment (sokra) [inline lib/dependencies/AMDDefineDependencyParserPlugin.js]: context arg not needed

File: lib/AsyncDependenciesBlock.js [hunk 5]
```diff
@@ -50,4 +50,4 @@ module.exports = class AsyncDependenciesBlock extends DependenciesBlock {
 			});
 		}
 	}
-}
+};
```
Comment (willmendesneto) [inline lib/AsyncDependenciesBlock.js]: This is unrelated to your changes. Could you rollback this, please?

**Comment by sokra** [comment]
Thanks
